src/ring_theory/artinian.lean

```diff
@@ -294,14 +325,15 @@ variables {N : Type w} [add_comm_group N] [module R N]
 --   exact submodule.bot_equiv_punit,
 -- end
 
-end
-
 /-- A ring is Artinian if it is Artinian as a module over itself.
 
 Strictly speaking, this should be called `is_left_artinian_ring` but we omit the `left_` for
 convenience in the commutative case. For a right Artinian ring, use `is_artinian Rᵐᵒᵖ R`. -/
 class is_artinian_ring (R) [ring R] extends is_artinian R R : Prop
 
+-- TODO: Can we define `is_artinian_ring` in a different way so this isn't needed?
+instance is_artinian_ring_of_fintype (R) [ring R] [fintype R] : is_artinian_ring R := ⟨⟩
```

Comment:
We already knows a module of finite cardinality is artinian, so I'm a bit surprised `extends` doesn't make this work automatically. If you define `@[reducible] def is_artinian_ring (R) [ring R] := is_artinian R R` then this does get automoatically inferred, but that deviates from `is_noetherian_ring`.

It seems this part of mathlib hasn't been adapted to use `finite`, so I think we should be satisfied with `fintype` now...

Reply:
Oops the linter complains, and it looks like we have to use `finite`. Also complains about priority; we could follow `is_artinian_of_fintype` to make it priority 100.

Reply:
I'll switch it to `finite`, but I think the reason it is done this way is because IIRC classes need to be `structure`s in Lean4, not defs. In lean4, though, we can create the instance `is_artinian_ring R ↔ is_artinian R R` (which would cause a loop right now) and solve this for free. I wonder what's the best way around this for now.

Reply:
Indeed it's changed from a def to a structure in #6028 in preparation for Lean 4. However there are still classes in mathlib that uses the `@[reducible] def` pattern, like [finite_dimensional](https://leanprover-community.github.io/mathlib_docs/linear_algebra/finite_dimensional.html#finite_dimensional) (defined to be `module.finite`). Should these all be changed to structures?